python/datafusion/dataframe.py

```diff
@@ -284,14 +324,41 @@ def join(
 
         Args:
             right: Other DataFrame to join with.
-            join_keys: Tuple of two lists of column names to join on.
+            on: Column names to join on in both dataframes.
             how: Type of join to perform. Supported types are "inner", "left",
                 "right", "full", "semi", "anti".
+            left_on: Join column of the left dataframe.
+            right_on: Join column of the right dataframe.
+            join_keys: Tuple of two lists of column names to join on. [Deprecated]
```

Comment:
With this addition, there are a set of mutually exclusive ways the user can call. You've done a very nice job of providing warnings when they are used incorrectly, but it would be very helpful to explain in the docstring that the user should provide either on OR both left_on and right_on.